src/Zoo.cpp

```diff
@@ -0,0 +1,261 @@
+#include "Zoo.h"
+
+void Zoo::stopAnimal(int row, int col) const
+{
+    Location l(row, col);
+    for (const auto& animal : animals) {
+        if (animal->getLocation() == l)
+        {
+            animal->stop();
+        }
+    }
+}
+
```

Comment:
why using the number of row and col to find the animal  and iterating over the array ??, you can pass the number of the animal and get directly to it. functions can be like do_something (int num)